lib/OpenQA/WebAPI/Controller/Admin/Workers.pm

```diff
@@ -109,5 +109,25 @@ sub previous_jobs_ajax {
     );
 }
 
+sub delete{
+    my ($self) = @_; 
+    my $w = $self->db->resultset('Workers')->find($self->param('worker_id'));
+
+    if ($w){
+        if ($w->status eq 'dead' && !$w->job){
+            $w->delete;
```

Comment:
Just because the subsequent message strongly suggests this operation will always succeed: That's not the case.

The deletion might fail if the row does not exists which can happen if it has been deleted since the `find(...)` (code might be executed in parallel in production!). However, I'm not sure how to handle this best. Maybe use a transaction or ignore errors by wrapping this into a try. @kraih What do you think?